src/signals.jl

```diff
@@ -129,6 +129,53 @@ Invoke/return `Legolas.write(path_or_io, signals, Schema("onda.signal@1"); kwarg
 """
 write_signals(path_or_io, signals; kwargs...) = Legolas.write(path_or_io, signals, Legolas.Schema("onda.signal@1"); kwargs...)
 
+"""
+    validate_signals(signals)
+
+Perform both table-level and row-level validation checks on the content of `signals`,
+a presumed `onda.signal` table. Returns `signals`.
+
+This function will throw an error in any of the following cases:
+
+- `Legolas.validate(signals, Legolas.Schema("onda.signal@1"))` throws an error
+- `Signal(row)` errors for any `row in Tables.rows(signals)`
+- `signals` contains rows with duplicate `file_path`s
+- `signals` contains rows that share a `recording` and `kind` but have overlapping `span`s
+"""
+function validate_signals(signals)
+    Legolas.validate(signals, Legolas.Schema("onda.signal@1"))
+    file_path_counts = Dict{Any,Int}()
+    spans_by_recording_and_kind = Dict{Tuple{UUID,String},Vector{TimeSpan}}()
+    for (i, row) in enumerate(Tables.rows(signals))
+        local signal
+        try
+            signal = Signal(row)
+        catch err
+            log("Encountered invalid row $i of given `onda.signal` table:")
+            rethrow(err)
+        end
```

Comment:
is this gonna catch duplicate channel names?

Reply:
We should add that check to the `Signal` constructor.

Unfortunately, it looks like the format spec doesn't actually disallow duplicate channel names - that's an oversight, certainly many downstream uses of Onda assume channel name uniqueness...

Technically the "right" path would be to have a separate PR that implements this restriction and bumps from v0.14 -> v0.15 and bumps from `onda.signal@1` -> `onda.signal@2`, etc. I'm verrrrrrrryyyyy tempted to sneak in a "technically breaking" patch, though - basically avoid the churn by assuming all valid downstream uses were already compliant with the technically-breaking change.

hhmmmmm

Reply:
we did a quick poll at Beacon and 4/4 responders voted we should just disallow this

Given that this is basically an oversight and i don't think would break any valid downstream use case in practice, I think we can treat this like a minor bug fix instead of a legit breaking change (even though I should note this is not technically The Right Way To Do Things in theory).

I'll include the change in this PR for sake of convenience.